Mathlib/CategoryTheory/Limits/Shapes/Pullback/CommSq.lean

```diff
@@ -201,6 +201,21 @@ theorem cone_snd (h : IsPullback fst snd f g) : h.cone.snd = snd :=
 noncomputable def isLimit (h : IsPullback fst snd f g) : IsLimit h.cone :=
   h.isLimit'.some
 
+/-- API for PullbackCone.IsLimit.lift for `IsPullback` -/
+noncomputable def lift (hP : IsPullback fst snd f g) {W : C} (h : W ⟶ X) (k : W ⟶ Y)
+    (w : h ≫ f = k ≫ g) : W ⟶ P :=
+  PullbackCone.IsLimit.lift hP.isLimit h k w
+
+@[reassoc (attr := simp)]
+lemma lift_fst (hP : IsPullback fst snd f g) {W : C} (h : W ⟶ X) (k : W ⟶ Y)
+    (w : h ≫ f = k ≫ g) : hP.lift h k w ≫ fst = h :=
+  PullbackCone.IsLimit.lift_fst hP.isLimit h k w
+
+@[reassoc (attr := simp)]
+lemma lift_snd (hP : IsPullback fst snd f g) {W : C} (h : W ⟶ X) (k : W ⟶ Y)
+    (w : h ≫ f = k ≫ g) : hP.lift h k w ≫ snd = k :=
+  PullbackCone.IsLimit.lift_snd hP.isLimit h k w
+
```

Comment:
Here, you may also add a `hom_ext` lemma: a morphism to `P` is determined by the two compositions. (Same for pushouts.)

Reply:
Fixed :)